tools/base/bazel_query.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python3
+"""Envoy Bazel query implementation.
+
+This module can be used either as a `py_binary` or a `py_library`.
+"""
+import pathlib
+import sys
+from functools import cached_property
+
+import abstracts
+
+from envoy.base.utils import ABazelQuery
+
+import envoy_repo
+
+
+@abstracts.implementer(ABazelQuery)
```

Comment:
This PR is clean, but it seems a bit magical. The reader doesn't know what ABazelQuery is, it's not in the repo etc. Also, where is the body of the implementation?

Not arguing at all with code structure, only with how the average reader is supposed to interpret and understand the code. What would a fresh reader do next after seeing this? How can we make the docs for ABazelQuery discoverable?

Reply:
yep, ill update with some info - ditto elsewhere

Reply:
updated above